src/app/service/event-template.service.ts

```diff
@@ -0,0 +1,116 @@
+import { Injectable } from '@angular/core';
+import { environment } from '../../environments/environment';
+import { BehaviorSubject, catchError, map, Observable, of } from 'rxjs';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { EventTemplate } from '../entity/EventTemplate';
+import EntityState from '../entity/EntityState';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EventTemplateService {
+  private data: EntityState<EventTemplate[]> = {
+    entity: [],
+    loading: false,
+    error: null,
+  };
+  private apiUrl: string = environment.baseUrl + '/v1/event-templates';
+  eventTemplateSubject = new BehaviorSubject<EntityState<EventTemplate[]>>(
+    this.data,
+  );
+
+  constructor(private _http: HttpClient) {
```

Comment:
I wouldn't use this notation (just my preference), I would just call this variable `http`, without the underscore. 

In case you want to use the underscore notation, also rename all private members using that notation (`data` abd `apiUrl`).